airflow/providers/amazon/aws/hooks/rds.py

```diff
@@ -48,3 +50,291 @@ class RdsHook(AwsGenericHook["RDSClient"]):
+            raise AirflowNotFoundException(e)
+        return response["DBClusterSnapshots"][0]["Status"].lower()
+
+    def wait_for_db_cluster_snapshot_state(
+        self, snapshot_id: str, target_state: str, check_interval: int = 30, max_attempts: int = 40
+    ) -> None:
+        """
+        Polls :py:meth:`RDS.Client.describe_db_cluster_snapshots` until the target state is reached.
+        An error is raised after a max number of attempts.
+
+        :param snapshot_id: The ID of the target DB cluster snapshot
+        :param target_state: Wait until this state is reached
+        :param check_interval: The amount of time in seconds to wait between attempts
+        :param max_attempts: The maximum number of attempts to be made
+
+        .. seealso::
+            A list of possible values for target_state:
+            https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds.html#RDS.Client.describe_db_cluster_snapshots
+        """
+
+        def poke():
+            return self.get_db_cluster_snapshot_state(snapshot_id)
+
+        target_state = target_state.lower()
+        if target_state in ("available", "deleted"):
+            waiter = self.conn.get_waiter(f"db_cluster_snapshot_{target_state}")
+            waiter.wait(
+                DBClusterSnapshotIdentifier=snapshot_id,
+                WaiterConfig={"Delay": check_interval, "MaxAttempts": max_attempts},
+            )
+        else:
+            self._wait_for_state(poke, target_state, check_interval, max_attempts)
+            self.log.info("DB cluster snapshot '%s' reached the '%s' state" % (snapshot_id, target_state))
+
+    def get_export_task_state(self, export_task_id: str) -> str:
+        """
+        Gets the current state of an RDS snapshot export to Amazon S3.
+
+        :param export_task_id: The identifier of the target snapshot export task.
+        :return: Returns the status of the snapshot export task as a string (eg. "canceled")
+        :rtype: str
+        :raises AirflowNotFoundException: If the export task does not exist.
+        """
+        try:
+            response = self.conn.describe_export_tasks(ExportTaskIdentifier=export_task_id)
+        # The RDS botocore documentation states that describe_export_tasks raises an exception of type ExportTaskNotFoundFault
+        # when the export task does not exist, but unit tests show that a generic ClientError is raised instead.
+        # https://botocore.amazonaws.com/v1/documentation/api/latest/reference/services/rds.html#RDS.Client.describe_export_tasks
+        except self.conn.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "ExportTaskNotFoundFault":
+                raise AirflowNotFoundException(e)
+            raise e
+        return response["ExportTasks"][0]["Status"].lower()
+
+    def wait_for_export_task_state(
+        self, export_task_id: str, target_state: str, check_interval: int = 30, max_attempts: int = 40
```

Comment:
The default boto waiter delay seems to be 30 for all waiters, but `max_attempts` varies from 40~60. I (arbitrarily) opted to go with 40 everywhere.